### PR TITLE
Update EIP 3074

### PR DESCRIPTION
**Pull request by jochem-brouwer**
Updates EIP-3074 (AUTHCALL) from https://github.com/ethereum/EIPs/commit/b59618ba071ef3c5912724262298833670f5c55a to https://github.com/ethereum/EIPs/commit/3c470e45eb794e2f995898741ec6624251455abc

- [x] https://github.com/ethereum/EIPs/commit/9ec9ff94da48fd72e32055b96805cb05aa1b8e74 + https://github.com/ethereum/EIPs/commit/94237c2ad0cc622e11bb0c8584fb499e64e59ff4 (re-align AUTH memory layout)
- [x] https://github.com/ethereum/EIPs/commit/10b9292d09dc01e9d1bd3dba264260e1e75d0788 (update MAGIC byte)
- [x] https://github.com/ethereum/EIPs/commit/4beea75a69df7bccc84b787c18c0a268bde80d31, https://github.com/ethereum/EIPs/commit/ed464603aa8647a16a416e7435adf4e95015dcc2 (add nonce revocation)
- [x] https://github.com/ethereum/EIPs/commit/eca4416ff3c025fcb6ec8cd4eac481e74e108481 (allow to send value from AUTHorized account)

~~WIP, because the spec is currently not clear about the nonce thing (can go two ways, have asked R&D)~~, nvm, I need more coffee today :sweat_smile:

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
The EIP got updated, please do not merge yet, I will implement the new spec.

File: packages/vm/test/api/EIPs/eip-3074-authcall.spec.ts [hunk 1]
```diff
@@ -636,7 +681,35 @@ describe('EIP-3074 AUTHCALL', () => {
     assert.equal(contractAccount!.balance, 2n, 'contract balance ok')
 
     const contractStorageAccount = await vm.stateManager.getAccount(contractStorageAddress)
-    assert.equal(contractStorageAccount!.balance, 1n, 'storage balance ok')
+    assert.equal(contractStorageAccount!.balance, 2n, 'storage balance ok')
+  })
+
+  it('Should throw is authorized account does not have enough balance', async () => {
```
Comment (acolytec3) [inline packages/vm/test/api/EIPs/eip-3074-authcall.spec.ts:687]:
```suggestion
  it('Should throw if authorized account does not have enough balance', async () => {
```

File: packages/evm/src/exceptions.ts [hunk 2]
```diff
@@ -24,8 +24,6 @@ export enum ERROR {
   INVALID_INPUT_LENGTH = 'invalid input length',
 
   AUTHCALL_UNSET = 'attempting to AUTHCALL without AUTH set',
-  AUTHCALL_NONZERO_VALUEEXT = 'attempting to execute AUTHCALL with nonzero external value',
-  AUTH_INVALID_S = 'invalid Signature: s-values greater than secp256k1n/2 are considered invalid',
```
Comment (acolytec3) [inline packages/evm/src/exceptions.ts:27]: Did I miss where this invalid S value issue is removed?

**Review by acolytec3 — CHANGES_REQUESTED** [review]
Spelling nit and a question, but seems to match up with the changes specified in the changes to the PR.

File: packages/evm/src/exceptions.ts (hunk 2 above)
Comment (jochem-brouwer) [reply packages/evm/src/exceptions.ts:27]: I had a discussion with lightclient on R&D and they said that this invalid s should push 0 on stack and not go out of gas.

**Review by acolytec3 — APPROVED** [review]

**Comment by holgerd77** [comment]
Ugh, already merged! 🙂 Congrats! 🎉

**Pull request merged**